src/incubator/WheelPicker/usePresenter.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {useRef} from 'react';
+import _ from 'lodash';
+import {ItemProps} from './Item';
+import useMiddleIndex from './helpers/useListMiddleIndex';
+
+type ItemValueTypes = ItemProps | number | string
+export { ItemValueTypes };
+
+type PropTypes = {
+    selectedValue: ItemValueTypes;
+    children?: JSX.Element | JSX.Element[];
+    items?: ItemProps[];
+    itemHeight: number;
+    preferredNumVisibleRows: number
+}
+
+type RowItem = {
+    value: string | number;
+    index: number
+}
+
+interface Presenter {
+    items: ItemProps[]
+    shouldControlComponent: (offset: number) => boolean;
+    index: number;
+    height: number;
+    getRowItemAtOffset: (offset: number) => RowItem;
+}
+
+const usePresenter = ({selectedValue, children, items: propItems, itemHeight, preferredNumVisibleRows}: PropTypes): Presenter => {
+  
+  const extractItemsFromChildren = (): ItemProps[] => {
+    const items = React.Children.map(children, child => {
+      let childAsType: ItemProps = {value: child?.props.value, label: child?.props.label};
+      return childAsType;
+    });
+    return items || [];
+  }
+
+  const items = useRef<ItemProps[]>(children ? extractItemsFromChildren() : propItems!).current
+  const middleIndex = useMiddleIndex({itemHeight, listSize: items.length});
+
+  const getSelectedValueIndex = (): number => {
+    if (_.isString(selectedValue) || _.isNumber(selectedValue)) {
+      return _.findIndex(items, {value: selectedValue});
+    }
+    return _.findIndex(items, {value: selectedValue?.value});
+  }
+
+  const shouldControlComponent = (offset: number): boolean => {
```

Comment:
at any point when swiping the wheel picker and `selectedValue` stays the same this will return true